### PR TITLE
Shader compilation

### PR DESCRIPTION
**Pull request by arose**
Reduce number of created programs/shaders
- Support specifying variants when creating graphics render-items
- Change double-side shader param from define to uniform
- Remove dMarkerType shader define (use uMarker as needed)
- Support to ignore defines depending on the shader variant
- Combine pickObject/pickInstance/pickGroup shader variants into one
- Combine markingDepth/markingMask shader variants into one
- Correctly set shader define flags for overpaint, transparency, substance, clipping

### CODE REVIEW

File: src/mol-canvas3d/passes/postprocessing.ts [hunk 1]
```diff
@@ -226,7 +226,7 @@ function getPostprocessingRenderable(ctx: WebGLContext, colorTexture: Texture, d
 
         uMaxPossibleViewZDiff: ValueCell.create(0.5),
 
-        dOcclusionEnable: ValueCell.create(false),
+        dOcclusionEnable: ValueCell.create(true),
```
Comment (dsehnal) [inline src/mol-canvas3d/passes/postprocessing.ts:229]: Why this change?

**Review by dsehnal — COMMENTED** [review]
Code looks good and it things seem to work after some basic testing.

File: src/mol-canvas3d/passes/postprocessing.ts (hunk 1 above)
Comment (arose) [reply src/mol-canvas3d/passes/postprocessing.ts:229]: That is just to bring it in line with the default params of the pass so the shader does not get re-created right after initialization. Btter, of course, would be to support passing the params but did not get to that.
